ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -291,15 +293,18 @@ def add_br_if_to_docker(self, bridge, ext_if, int_if):
 
         return
 
-    def add_ip_to_docker_if(self, int_if, mgmt_ip_addr, mgmt_ipv6_addr=None, mgmt_gw=None):
+    def add_ip_to_docker_if(self, int_if, mgmt_ip_addr, mgmt_ipv6_addr=None, mgmt_gw=None, mgmt_gw_v6=None):
         self.update()
         if int_if in self.cntr_ifaces:
             VMTopology.cmd("nsenter -t %s -n ip addr flush dev %s" % (self.pid, int_if))
             VMTopology.cmd("nsenter -t %s -n ip addr add %s dev %s" % (self.pid, mgmt_ip_addr, int_if))
-            if mgmt_ipv6_addr:
-                VMTopology.cmd("nsenter -t %s -n ip -6 addr add %s dev %s" % (self.pid, mgmt_ipv6_addr, int_if))
             if mgmt_gw:
                 VMTopology.cmd("nsenter -t %s -n ip route add default via %s dev %s" % (self.pid, mgmt_gw, int_if))
+            if mgmt_ipv6_addr and mgmt_gw_v6:
+                VMTopology.cmd("nsenter -t %s -n ip -6 addr flush dev %s" % (self.pid, int_if))
+                VMTopology.cmd("nsenter -t %s -n ip -6 route flush default" % (self.pid))
```

Comment:
I think before we saw an issue where these commands would fail if the the gateway route was not present - does this new code address that? Are there any issues if we issue these steps inside a totally new setup where no routes are present yet?

Reply:
Yes, flush will not return an error if ipv6 address or route was not previously configured.
root@8402ee025f72:~# route -n -6 | grep ::/0   
::/0                           ::                         !n   -1  1     0 lo
::/0                           ::                         !n   -1  1     0 lo
root@8402ee025f72:~# ip -6 route flush default ---------------------- does not return an error
root@8402ee025f72:~# ip -6 route add default via fec0::ffff:afa:1
root@8402ee025f72:~# route -n -6 | grep ::/0                     
::/0                           fec0::ffff:afa:1           UG   1024 1     0 mgmt
::/0                           ::                         !n   -1  1     0 lo
root@8402ee025f72:~# ip -6 route flush default    ------------------------------- removes ipv6 default route             
root@8402ee025f72:~# route -n -6 | grep ::/0                     
::/0                           ::                         !n   -1  1     0 lo
::/0                           ::                         !n   -1  1     0 lo

Reply:
Sorry, didn't follow this issue closely before. What was the error before? Why it is not failing now? Did you change the command or parameter?

Reply:
Earlier, I added route to ipv6 mgmt gw without flushing the existing default route. That caused RTNETLINK existing route error when doing a refresh-dut on an existing testbed. We cannot merely delete the route as deletion of non existing route during add-topo will cause a different error. Hence, fix is to flush ipv6 address and route specifically.
In case of IPv4, "ip addr flush dev" removes the ipv4 address as well as the default route. But for ipv6, we have to specifically flush the ip address and default route.